code/modules/reagents/Chemistry-Reagents/Medicine-Reagents.dm

```diff
@@ -468,13 +468,23 @@
 	metabolization_rate = 0.5 * REAGENTS_METABOLISM
 	overdose_threshold = 30
 	addiction_threshold = 30
+	stun_threshold = 6
 
 /datum/reagent/medicine/ephedrine/on_mob_life(mob/living/M)
 	M.status_flags |= GOTTAGOFAST
-	M.AdjustParalysis(-0.4)
-	M.AdjustStunned(-0.4)
-	M.AdjustWeakened(-0.4)
-	M.adjustStaminaLoss(-0.5*REM)
+	if(!(M.stunned || M.weakened || M.paralysis))
+		stun_timer++
+		metabolization_rate = 0.5* REAGENTS_METABOLISM
+		M.adjustStaminaLoss(-1)
+	else
+		metabolization_rate = 2 * REAGENTS_METABOLISM
```

Comment:
Could you not use initial(metabolization_rate) is your multiplier?
So if someone wants to change the metabolizationrate, they only need to change the value at the definitions of the reagent?